### ISSUE TITLE
List module routes not always handled properly

### ISSUE THREAD

**Issue by ric2016** [fisharebest/webtrees#3171]
For private trees, the list urls result in errors when the user is logged out, e.g. when using [/webtrees/module/source_list/List/tree1](/webtrees/module/source_list/List/tree1):

```assert(): assert($tree instanceof Tree) failed …\app\Module\SourceListModule.php:84
#0 [internal function]: Fisharebest\Webtrees\Webtrees::Fisharebest\Webtrees\{closure}(2, 'assert(): asser...', 'C:\\xampp7\\htdoc...', 84, Array)
#1 …\app\Module\SourceListModule.php(84): assert(false, 'assert($tree in...')
#2 …\app\Http\RequestHandlers\ModuleAction.php(90): Fisharebest\Webtrees\Module\SourceListModule->getListAction(Object(Nyholm\Psr7\ServerRequest))
```

In general, [Router.php handles these cases](https://github.com/fisharebest/webtrees/blob/5c20d904a5240e21cf5895bd3d43ed728a641d79/app/Http/Middleware/Router.php#L120), but only when the tree parameter is explicitly mandatory, which isn't the case for module routes.

Therefore, the respective check and redirect has to be made either by the module itself, or we'll have to define special module routes with mandatory tree parameters (we had discussed this a while ago in a different context [here](https://www.webtrees.net/index.php/en/forum/2-open-discussion/33687-pretty-urls-in-2-x#72866)), which would solve this as a side effect.

**Comment by fisharebest** [comment]
This was fixed (yesterday) by #3169

**Comment by fisharebest** [comment]
Correction - a similar problem was fixed in #3169.
